src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java

```diff
@@ -131,4 +173,167 @@ public void joinCommonsTest() throws Exception {
+        .commonsId(2L)
+        .totalWealth(0)
+        .build();
+
+    String requestBody = mapper.writeValueAsString(uc);
+
+    when(userCommonsRepository.findByCommonsIdAndUserId(anyLong(),anyLong())).thenReturn(Optional.empty());
+    when(userCommonsRepository.save(eq(uc))).thenReturn(ucSaved);
+    when(commonsRepository.findById(eq(2L))).thenReturn(Optional.empty());
+
+    MvcResult response = mockMvc
+        .perform(post("/api/commons/join?commonsId=2").with(csrf()).contentType(MediaType.APPLICATION_JSON)
+            .characterEncoding("utf-8").content(requestBody))
+        .andExpect(status().is(404)).andReturn();
+
+    verify(userCommonsRepository, times(1)).findByCommonsIdAndUserId(2L, 1L);
+    verify(userCommonsRepository, times(1)).save(uc);
+
+    String responseString = response.getResponse().getContentAsString();
+
+    if(responseString.charAt(2) == 'm'){
+      assertEquals("{\"message\":\"Commons with id 2 not found\",\"type\":\"EntityNotFoundException\"}", responseString); 
+    }else{
+      assertEquals("{\"type\":\"EntityNotFoundException\",\"message\":\"Commons with id 2 not found\"}", responseString); 
+    }
+  }
+
+  @WithMockUser(roles = {"ADMIN"})
+  @Test
+  public void deleteUserFromCommonsTest() throws Exception {
+    UserCommons uc = UserCommons.builder()
+        .id(16L)
+        .userId(1L)
+        .commonsId(2L)
+        .totalWealth(0)
+        .build();
+
+    String requestBody = mapper.writeValueAsString(uc);
+
+    when(userCommonsRepository.findByCommonsIdAndUserId(2L,1L)).thenReturn(Optional.of(uc));
+
+    MvcResult response = mockMvc
+        .perform(delete("/api/commons/2/users/1").with(csrf()).contentType(MediaType.APPLICATION_JSON)
+            .characterEncoding("utf-8").content(requestBody))
+        .andExpect(status().is(204)).andReturn();
+    
+    verify(userCommonsRepository, times(1)).findByCommonsIdAndUserId(2L, 1L);
+    verify(userCommonsRepository, times(1)).deleteById(16L);
+    //verify(userCommonsRepository, times(1)).deleteById(1L);
+
+    String responseString = response.getResponse().getContentAsString();
+
+    assertEquals(responseString, "");
+  }
+
+  @WithMockUser(roles = {"ADMIN"})
+  @Test
+  public void deleteUserFromCommonsTest_nonexistent_userCommons() throws Exception {
+    UserCommons uc = UserCommons.builder()
+        .id(16L)
+        .userId(1L)
+        .commonsId(2L)
+        .totalWealth(0)
+        .build();
+
+    String requestBody = mapper.writeValueAsString(uc);
+
+    when(userCommonsRepository.findByCommonsIdAndUserId(2L,1L)).thenReturn(Optional.empty());
+
+    MvcResult response;
+    try{
+      response = mockMvc
+        .perform(delete("/api/commons/2/users/1").with(csrf()).contentType(MediaType.APPLICATION_JSON)
+            .characterEncoding("utf-8").content(requestBody))
+        .andExpect(status().is(204)).andReturn();
+
+    //The way this works is very interesting. The error message is sent as the value of a nested exception.
```

Comment:
That's definitely strange! Not critical to resolve in this PR, but it's a good thing for us to keep a note of.

Reply:
This isn't great. Why is the back end throwing a raw `Exception`? It should be doing HTTP stuff, not crashing itself.

Reply:
Yeah, wasn't sure how to workaround it. Thanks for the acknowledgement!